langchain/embeddings/gpt4all.py

```diff
@@ -0,0 +1,61 @@
+"""Wrapper around GPT4All embedding models."""
+from typing import Any, Dict, List
+
+from pydantic import BaseModel, root_validator
+
+from langchain.embeddings.base import Embeddings
+
+
+class GPT4AllEmbeddings(BaseModel, Embeddings):
+    """Wrapper around GPT4All embedding models.
+
+    To use, you should have the gpt4all python package installed
+
+    Example:
+        .. code-block:: python
+            from langchain.embeddings import GPT4AllEmbeddings
+            gpt4all_embd = GPT4AllEmbeddings()
+    """
+
+    client: Any  #: :meta private:
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that GPT4All library is installed."""
+
+        try:
+            from gpt4all import Embed4All
+
+            values["client"] = Embed4All()
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Could not import gpt4all library. "
+                "Please install the gpt4all library to "
+                "use this embedding model: pip install gpt4all"
+            )
+        return values
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        """Embed a list of documents using GPT4All.
+
+        Args:
+            texts: The list of texts to embed.
+
+        Returns:
+            List of embeddings, one for each text.
+        """
+
+        embeddings = [self.client.embed(text) for text in texts]
+        return [list(map(float, e)) for e in embeddings]
```

Comment:
ooc why is this needed? Embed4All.embed signature suggests this already returns List[float]